### PR TITLE
Fixed sonar issues in identity-metadata-saml2

### PR DESCRIPTION
**Pull request by supunharsha**

### CODE REVIEW

File: components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java [hunk 1]
```diff
@@ -206,9 +206,9 @@ private static FederatedAuthenticatorConfig parse(EntityDescriptor entityDescrip
                                 } catch (Exception ex) {
                                     log.error("Error !!!!", ex);
                                 }
-                                if (use != null && use.equals("SIGNING")) {
+                                if (use.equals("SIGNING")) {
```
Comment (GayanM) [inline components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java]: constant must be first and define variables for constants

File: components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java [hunk 2]
```diff
@@ -206,9 +206,9 @@ private static FederatedAuthenticatorConfig parse(EntityDescriptor entityDescrip
...
                                     properties[10].setValue("true");
-                                } else if (use != null && use.equals("ENCRYPTION")) {
+                                } else if (use.equals("ENCRYPTION")) {
```
Comment (GayanM) [inline components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java]: constant must be first and define variables for constants

**Review by GayanM — APPROVED** [review]

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ private void setNameIDFormat(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProv
     }
 
     private void setClaims(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProviderDO samlssoServiceProviderDO) {
-        List<AttributeConsumingService> services = new ArrayList<>();
+        List<AttributeConsumingService> services;
```
Comment (madumalt) [inline components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java:106]: can merge declaration and intialization

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java [hunk 4]
```diff
@@ -103,7 +103,7 @@ private void setNameIDFormat(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProv
...
         services = spssoDescriptor.getAttributeConsumingServices();
         if (services != null && services.size() > 0) {
```
Comment (madumalt) [inline components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java:108]: use CollectionUtils.isNotEmpty

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java [hunk 5]
```diff
@@ -103,7 +103,7 @@ private void setNameIDFormat(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProv
...
             //assuming that only one AttrbuteComsumingIndex exists
```
Comment (madumalt) [inline components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java:109]: The following for loop is does nothing, so no use. Else clause is also of no use.
Basically, this whole method is doing nothing useful.
Either can remove this method if there are no usages, 
or should refine the method such that it does the intended work

**Review by rushmin — APPROVED** [review]

**Pull request merged**